src/manager.ts

```diff
@@ -0,0 +1,79 @@
+import { Identity } from '@semaphore-protocol/identity'
+import RLN from './rln'
+import Registry from './registry'
+import Cache, { Status } from './cache'
+import { StrBigInt, VerificationKey } from './types'
+import poseidon from 'poseidon-lite'
+
+export default class RLNManager {
+  identity: Identity
+
+  rlnIdentifier: bigint
+
+  treeDepth: number
+
+  zeroValue: bigint
+
+  rlnInstance: RLN
+
+  registryInstance: Registry
+
+  cacheInstance: Cache
+
+  constructor(
+    wasmFilePath: string,
+    finalZkeyPath: string,
+    verificationKey: VerificationKey,
+    rlnIdentifier: bigint,
+    identity: string,
```

Comment:
Not sure if I missed any context: I'm wondering if we could change `identity` to type `Identity` since this might save us a `identity.toString()`, i.e., we can have `new RLNManager(..., identity, ...)` instead of `new RLNManager(..., identity.toString(), ...)`.
```suggestion
    identity?: string,
```

Reply:
Ya I've been thinking about how to handle the identity, and I think it might be good to force the user to provide a semaphore identity. We could provide a method to generate one too, but I want it to be very clear that they have to provide an identity. If we generate a new one every instance, they may not understand how they identity is actually used.